packages/tpetra/core/example/Finite-Element-Assembly/fem_assembly_Element.hpp

```diff
@@ -98,6 +99,13 @@ void ReferenceQuad4RHS(Teuchos::Array<Scalar>& rhs) {
   std::fill(rhs.begin(), rhs.end(), static_cast<Scalar>(.25));
 }
 
+template<class ViewType>
+KOKKOS_INLINE_FUNCTION void ReferenceQuad4RHS_Kokkos(ViewType& rhs) {
+  for(size_t i=0; i<rhs.extent(0); i++)
```

Comment:
`ReferenceQuad4RHS` resizes, but `ReferenceQuad4RHS_Kokkos` does not.  I would say, do the resizing outside, and use overloading instead of a different function name.
```c++
KOKKOS_INLINE_FUNCTION template<class ViewType>
void referenceQuad4RHS (ViewType rhs) {
  rhs[0] = 0.25;
  rhs[1] = 0.25;
  rhs[2] = 0.25;
  rhs[3] = 0.25;
}

// this assumes that nvcc permits overloads 
// that have different __device__ markings.
void referenceQuad4RHS (Teuchos::ArrayView<double> rhs) {
  rhs[0] = 0.25;
  rhs[1] = 0.25;
  rhs[2] = 0.25;
  rhs[3] = 0.25;
}
```

Reply:
The resize was an artifact of the original code (no good idea why it was there).  Easily fixed, though.